src/data/markdown/translated-guides/en/03 Results output/100 End-of-test/150-custom-summary.md

```diff
@@ -1,35 +1,76 @@
 ---
 title: Custom summary
-excerpt: With the handle summary, you can customize every part of your report, 
+excerpt: With the handlesummary() function, you can customize every part of your report. Change the content, redirect output, and more.
 ---
 
-Use the `handleSummary()` function to completely customize the end-of-test summary report.
-
-k6 calls`handleSummary()` at the end of the test run, even after [`teardown()`](/using-k6/test-life-cycle).
-If `handleSummary()` is exported, k6 does _not_ print the default summary.
-
-Besides customizing the CLI summary, you can also transform the summary data into machine- or human-readable formats.
-This lets you create JS-helper functions that generate JSON, CSV, XML (JUnit/xUnit/etc.), HTML, etc. files from the summary data.
+With the `handleSummary()` callback, you can completely customize your end-of-test summary.
+In this document, you can read about:
+- How `handleSummary()` works
+- How to customize the content and output location of your summary
+- The data structure of the summary object
 
 <Blockquote mod="note"
-title="handleSummary() is available on only local tests."
->
+title="handleSummary() is available only for local tests.">
 
-However we plan to support the feature for k6 Cloud tests, too.
+However, we plan to support the feature for k6 Cloud tests, too.
 [Track progress in this issue](https://github.com/grafana/k6-cloud-feature-requests/issues/24).
 
 </Blockquote>
 
-## Data format returned by `handleSummary()`
+## About the `handleSummary()` callback
+
+After your VU code runs, k6 aggregates your metrics into an object.
+The `handleSummary()` function takes this object as an argument (called `data` in all examples here).
+k6 calls `handleSummary()` at the end of the test run, even after [`teardown()`](/using-k6/test-life-cycle).
```

Comment:
> k6 calls `handleSummary()` at the end of the test run, even after [`teardown()`](/using-k6/test-life-cycle).

I think this could be explained later.